src/analysis/mesh/qgsmeshcalcutils.cpp

```diff
@@ -699,20 +700,9 @@ void QgsMeshCalcUtils::updateMesh() const
 {
   if ( ! mMeshLayer->nativeMesh() )
   {
-    // THIS code is very confusing -- someone please add some explanation as to why a map renderer is created here! (Or better,
-    // add explicit members to do whatever it is that's actually wanted here, instead of creating the map renderer)
-
-    // we do not care about triangles,
-    // we just want transformed coordinates
-    // of the native mesh. So create
-    // some dummy triangular mesh.
-    QgsMapSettings mapSettings;
-    mapSettings.setExtent( mMeshLayer->extent() );
-    mapSettings.setDestinationCrs( mMeshLayer->crs() );
-    mapSettings.setOutputDpi( 96 );
-    QgsRenderContext context = QgsRenderContext::fromMapSettings( mapSettings );
-
-    delete mMeshLayer->createMapRenderer( context );
+    QgsCoordinateTransform transform( mMeshLayer->crs(), mMeshLayer->crs(), QgsProject::instance() );
```

Comment:
You need to avoid the QgsProject::instance() call here -- that will cause all sorts of issues in processing and elsewhere. You'll need to instead make the QgsTransformContext an argument for updateMesh() (or in the constructor for QgsMeshCalcUtils )

Reply:
I am asking my self, here, not sure there is a real context for transform. More over, there is not a real transform as the transform is from layer crs to layer crs. It is just a workaround to create a triangular mesh in the layer crs taht will be used to mesh calculator operation. The CRS will always be the layer CRS.
Can we just used a "empty" QgsTransformContext obtained by default constructor ?

Reply:
in that case maybe you can create a function that does not take transformation at all (or as default transformation is invalid) and it will not reproject vertices when creating the triangular mesh. And here in the code you will not pass there any context ,so it would be obvious what is going on here.

Reply:
@PeterPetrik , very good idea, so no need the transform here and no workaround anymore!